abra_core/src/typechecker/typechecker.rs

```diff
@@ -80,59 +81,43 @@ impl Typechecker {
         scope.bindings.insert(name.to_string(), binding);
     }
 
-    fn update_binding<F>(&mut self, name: &String, func: F)
-        where F: FnOnce(&mut ScopeBinding)
-    {
-        for scope in self.scopes.iter_mut().rev() {
-            for (binding_name, entry) in scope.bindings.iter_mut() {
-                if binding_name == name {
-                    return func(entry);
-                }
-            }
-        }
-    }
-
     fn get_binding_mut(&mut self, name: &str) -> Option<&mut ScopeBinding> {
```

Comment:
I (possibly unnecessarily) "cleaned up" these methods a bit, to remove the nested for-loops and instead used (abused?) iterators a bit. _I_ think it's easier to read...